app/application.py

```diff
@@ -55,6 +55,28 @@ def drop_database():
 @app.before_request
 def before_request():
     if request.endpoint is not None and 'health' not in request.endpoint and request.method != 'OPTIONS':
+        header = request.headers
```

Comment:
Can you split your code into a private method called _log_request and call it before the top if statement. Thanks